ArduPlane/commands_logic.cpp

```diff
@@ -526,17 +526,33 @@ bool Plane::verify_takeoff()
             // course. This keeps wings level until we are ready to
             // rotate, and also allows us to cope with arbitrary
             // compass errors for auto takeoff
-            float takeoff_course = wrap_PI(radians(gps.ground_course_cd()*0.01f)) - steer_state.locked_course_err;
-            takeoff_course = wrap_PI(takeoff_course);
-            steer_state.hold_course_cd = wrap_360_cd(degrees(takeoff_course)*100);
+            const float ground_course_rad = wrap_PI(ahrs.groundspeed_vector().angle());
+            const float ground_course = wrap_360(degrees(ground_course_rad));
+            const float takeoff_course_rad = wrap_PI(ground_course_rad - steer_state.locked_course_err);
+
+            steer_state.hold_course_cd = wrap_360_cd(degrees(takeoff_course_rad)*100);
+
+            // when true, we will follow the recorded takeoff heading.
+            // when false, we xtrack to a virtual waypoint 10km ahead of us on this takeoff heading
+            // TODO: turn this hardcoded value into a param/option?
+            steer_state.locked_course = false;
+
+            next_WP_loc = current_loc;
+            next_WP_loc.offset_bearing(ground_course, 10000); // push it out 10km
```

Comment:
can not manipulate next_WP_loc here because set_next_WP() sets it. Must use a temp local loc instead.

Reply:
this course should be the exact same as steer_state.hold_course_cd. That will also reduce some of the above maths